pkg/queryfrontend/queryrange_codec.go

```diff
@@ -262,3 +294,16 @@ func matchersToStringSlice(storeMatchers [][]*labels.Matcher) []string {
 	}
 	return res
 }
+
+func encodeShardInfo(info *storepb.ShardInfo) (string, error) {
```

Comment:
This might not be ideal because it will be hard to set the parameter manually in the URL. We can start like this and optimize later, or we can come up with something more user friendly.